testcontainers/src/runners/async_runner.rs

```diff
@@ -384,6 +384,40 @@ mod tests {
         assert!(!client.network_exists("awesome-net-2").await)
     }
 
+    #[tokio::test]
+    async fn async_should_rely_on_network_mode_when_network_is_provided_and_settings_bridge_empty()
```

Comment:
Hi @dlen!
I was about to merge the PR, but wanted to double-check the test coverage and current behavior in `main`. 
And understood that these tests are passing against it.

I'm just wondering if this could be covered better, because if we change this back - we don't see the crash, however we are aware of some issues.
Sorry for the late notice